### PR TITLE
PARQUET-1193: [CPP] Implement ColumnOrder to support min_value and max_value

### PR DESCRIPTION
**Pull request by majetideepak**
Changes:

1. Update parquet.thrift format
2. Add ColumnOrder Implementation
3. Make Int96 sort order UNKNOWN

### CODE REVIEW

File: src/parquet/statistics-test.cc [hunk 1]
```diff
@@ -397,7 +397,7 @@ TEST(CorrectStatistics, Basics) {
   ASSERT_TRUE(column_chunk4->is_stats_set());
   auto column_chunk5 = ColumnChunkMetaData::Make(
       reinterpret_cast<const uint8_t*>(&col_chunk), schema.Column(4), &version);
-  ASSERT_TRUE(column_chunk5->is_stats_set());
+  ASSERT_FALSE(column_chunk5->is_stats_set());
```
Comment (xhochy) [inline src/parquet/statistics-test.cc]: Why is this changed needed here?

File: src/parquet/types.h [hunk 2]
```diff
@@ -130,6 +130,21 @@ struct SortOrder {
   enum type { SIGNED, UNSIGNED, UNKNOWN };
 };
 
+class ColumnOrder {
+ public:
```
Comment (xhochy) [inline src/parquet/types.h]: From where stems the need to have a full class for the column order? From looking at the code, it seems to me like using the enum should be sufficient.

File: src/parquet/statistics-test.cc (hunk 1 above)
Comment (majetideepak) [reply src/parquet/statistics-test.cc]: `column_chunk5` is defined for `LogicalType::INTERVAL`, which has sort order `UNKNOWN`.
I modified this PR to not set stats for all `UNKNOWN` sort orders.

File: src/parquet/types.h (hunk 2 above)
Comment (majetideepak) [reply src/parquet/types.h]: Yes, enum is sufficient for this PR. I added the full class for further extensions. The Java implementation did something similar: 
https://github.com/apache/parquet-mr/pull/435/files#diff-37183c50b2eaab3ca048f9c5954e5b52R26
